cfme/tests/ansible/test_embedded_ansible_automate.py

```diff
@@ -418,3 +420,60 @@ def test_variable_pass(request, appliance, setup_ansible_repository, import_data
         provision_request = appliance.collections.requests.instantiate(request_description)
         provision_request.wait_for_request(method="ui")
         request.addfinalizer(provision_request.remove_request)
+
+
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream("5.10")
+@pytest.mark.meta(automates=[1677575])
+@pytest.mark.parametrize(
+    "import_data",
+    [DatastoreImport("bz_1677575.zip", "bz_1677575", None)],
+    ids=["datastore"],
+)
+def test_import_domain_containing_playbook_method(request, appliance, setup_ansible_repository,
+                                                  import_data):
+    """This test case tests support of Export/Import of Domain with Ansible Method
+
+    Bugzilla:
+        1677575
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.11
+        casecomponent: Automate
+        tags: automate
+        setup:
+            1. Add playbook repository
+            2. Create a new automate method with playbook type.
+            3. Fill the required fields, for instance repository and playbook.
+            4. Export this datastore.
+            5. Playbook method fields are stored as a names instead of IDs. (this is not
+               possible via automate need to check manually in method yaml)
+        testSteps:
+            1. Import the exported datastore and change name of playbook in method yaml to invalid
+               playbook name(Note: These test steps needs to execute manually and then import
+               datastore)
+        expectedResults:
+            1. Proper error should be displayed while importing datastore with invalid playbook
+    """
+    # Download datastore file from FTP server
+    fs = FTPClientWrapper(cfme_data.ftpserver.entities.datastores)
+    file_path = fs.download(import_data.file_name)
+
+    # Import datastore file to appliance
+    datastore = appliance.collections.automate_import_exports.instantiate(
+        import_type="file", file_path=file_path
+    )
+    domain = datastore.import_domain_from(import_data.from_domain, import_data.to_domain)
+    request.addfinalizer(domain.delete_if_exists)
+    view = appliance.browser.create_view(FileImportSelectorView)
+    # setup_ansible_repository.name is ansible repository name already there in datastore yml which
+    # we are going to import.
+    error_msg = (
+        f"Playbook 'invalid_1677575.yml' not found in repository '{setup_ansible_repository.name}'"
+    )
+    assert error_msg in view.flash.read()[0]
```

Comment:
Can you use `view.flash.messages`? 

```suggestion
    assert error_msg in view.flash.messages
```

Reply:
Hey @john-dupuy Here we are getting one message.
- By using ```messages``` we are only getting object of flash message in list:
```
ipdb> view.flash.messages                                                                                                                                                    
[<widgetastic_manageiq.V2VFlashMessage object at 0x7ff493d76b90>]

```
- So if we used as you suggested then we need to change it like:
```
ipdb> assert error_msg in view.flash.messages[0].read()                                                                                                                      
```
Hence keeping as it is.

Reply:
Does `view.flash.assert_message(message)` work for error messages? Have you tried it?

Reply:
So @valaparthvi 
I am not checking exact whole flash message which is nearly 3-4 line long. I am asserting for error message; I need for this test case. So I can not use **assert_message**. 
Please let me know, if you have any other view.

Reply:
So as discussed offline with @valaparthvi .
Tried with ```partial=True``` and ```assert_message``` which worked as expected.

Reply:
@tpapaioa is also currently working on supporting regex compiled expressions for assert_message, which would be useful here. Though I don't want to block your PR on that enhancement - you can change from partial to regex once its released if you'd like - at a later time.